jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -1200,6 +1200,20 @@ YUI.add('juju-env-api', function(Y) {
         return;
       }
       const ecs = this.get('ecs');
+      if (args.charmResources) {
```

Comment:
I think callers have the responsibility to do this, and injecting an extraneous deploy argument like this can be confusing.